src/classes/RD2_OpportunityService.cls

```diff
@@ -123,16 +123,19 @@ public with sharing class RD2_OpportunityService {
     */
     private Opportunity buildOpportunity(npe03__Recurring_Donation__c rd, Date closeDate) {
 
+        Integer installmentNumber = Integer.valueOf((rd.npe03__Total_Paid_Installments__c != null ?
```

Comment:
Though this can work, I'd like to make one suggestion for a different approach to solving the problem. Basically along the lines of the following:

```
Opportunity opp = new Opportunity(
    ... 
    CloseDate = closeDate, 
    Recurring_Donation_Installment_Number__c = (rd.npe03__Total_Paid_Installments__c != null ?
        rd.npe03__Total_Paid_Installments__c + 1 : 1)
     ...
);
opp.Name = buildOpportunityName(rd);
```

In the above, the Opp instance is being built out without Name. Then the `buildOpportunityName()` method can be very simple with a single parameter for the recurring donation. 

What do you think?

Reply:
Thanks, @force2b , I thought about this approach too and not sure which to pick, I will change it as per your suggestion.

Reply:
@force2b , buildopportunityname might need another parameter which is the opportunity. I have made the changes.